src/OSPSuite.Infrastructure.Import/Core/Mappers/IDataSetToDataRepositoryMapper.cs

```diff
@@ -0,0 +1,203 @@
+                        if (dataColumn.Values[i] < 0F)
+                        {
+                           dataColumn[i] = float.NaN;
+                        }
+                     break;
+                  case Constants.STD_DEV_GEOMETRIC:
+                     errorType = AuxiliaryType.GeometricStdDev;
+                     for (i = 0; i < dataColumn.Values.Count; i++)
+                        if (dataColumn.Values[i] < 1F)
+                        {
+                           dataColumn[i] = float.NaN;
+                        }
+                     break;
+               }
+            }
+            propInfo.SetValue(dataColumn.DataInfo, errorType, null);
+         }
+
+         //special case: Origin
+         //if AuxiliaryType is set, set Origin to ObservationAuxiliary else to Observation for standard columns
+         //type is set to BaseGrid for baseGrid column
+         if (dataColumn.IsBaseGrid())
+            dataInfo.Origin = ColumnOrigins.BaseGrid;
+         else
+            dataInfo.Origin = (dataColumn.DataInfo.AuxiliaryType == AuxiliaryType.Undefined)
+               ? ColumnOrigins.Observation
+               : ColumnOrigins.ObservationAuxiliary;
+
+         //meta data information and input parameters currently not handled
+         dataRepository.Add(dataColumn);
+      }
+
+      private static DataColumn findColumnByName(IEnumerable<DataColumn> columns, string name)
+      {
+         var column = columns.FirstOrDefault(col => col.Name == name);
+         if (column == null)
+            throw new Exception(name);//this should actually be named
+         return column;
+      }
+      private static void addExtendedPropertyForSource(string fileName, string sheetName,  DataRepository dataRepository)
+      {
+         if (!string.IsNullOrEmpty(fileName))
+         {
+            var sourceProperty =
+               Activator.CreateInstance(typeof(ExtendedProperty<>).MakeGenericType(fileName.GetType()))
+                  as IExtendedProperty;
+            if (sourceProperty != null)
+            {
+               sourceProperty.Name = "Source";
+               sourceProperty.ValueAsObject = fileName;
+               dataRepository.ExtendedProperties.Add(sourceProperty);
+            }
+
+            dataRepository.ExtendedProperties.Add(new ExtendedProperty<string> { Name = Constants.FILE, Value = fileName });
+            dataRepository.ExtendedProperties.Add(new ExtendedProperty<string> { Name = Constants.SHEET, Value = sheetName });
+         }
+      }
+
+      private static bool containsColumnByName(IEnumerable<DataColumn> columns, string name)
+      {
+         return columns.Any(col => col.Name == name);
+      }
+
+      private ParsedDataSet getDataSet(IDataSource dataSource, int dataSetIndex, out int sheetIndex)
+      {
+         sheetIndex = 0;
+         var sheet = dataSource.DataSets.GetEnumerator();
+         while (sheet.MoveNext() && dataSetIndex >= 0)
```

Comment:
What is going on here? 
Why is there a return value AND and out parameter?

Reply:
out parameter changed to become second return value